### ISSUE TITLE
Entries appear multiple times

### ISSUE THREAD

**Issue by tuximail** [stefan-niedermann/nextcloud-deck#612]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**
When I add cards to a board in a high frequent manner (e.g. Multiple cards in a row, with switching back and forth to the app), single cards appear multiple times after they synced. Sometimes one card can be seen 3-5 times.

**To reproduce**
Issue is reproducible with an account from [try.nextcloud.com](https://try.nextcloud.com):
- [ ] Yes
- [x] No

Create cards with a high frequency. Maybe also switch between the app and another in between

Additional Information:
The Bug is observed on a shared board, but is also reproducable on a board only used by one person.

**Expected behavior**
Cards should appear only ones

**Versions**
 - Nextcloud: 18.0.7
 - Nextcloud Deck: 1.0.5
 - Nextcloud Android: 20200714
 - Nextcloud Android Deck: 1.8.2

**Smartphone (please complete the following information):**
- Device: LG G7
- Android-Version: 9
- App-Store:
  - [ ] Google Play Store
  - [x] F-Droid

**Possible fix**
Maybe it would be enough if the app would check if a card with the same content has been sent to the backend in the last seconds and if so, ommit it.

**Comment by stefan-niedermann** [comment]
Thank you for the report. Please fill in the **complete** issue template, otherwise we can not work on it. It is there for a reason 🙂.

**Comment by tuximail** [comment]
I am really sorry for the incomplete report. I was submitting it via mobile with disabled javascript and as I just found out the templates appear not in the issue textfield in that scenario..
Updated all missing fields in the original post.

**Comment by desperateCoder** [comment]
No JS? I think in that case @stefan-niedermann can't really be mad 🤣

**Comment by stefan-niedermann** [comment]
@desperateCoder might this be a symptome of parallel syncs maybe?

@tuximail thanks for adding the infos 🙂

**Comment by desperateCoder** [comment]
@stefan-niedermann most likely not... Especially in the online case: it is a completely isolated process, only linked to the rest of the application by the one IResponseCallback you create on the 'save'-tap... If you can promise you don't call it more than once, let's say on destroy or whatever, this is totally a weird one. I'll have to add some logs to verify, where exactly some process goes crazy...

**Comment by tuximail** [comment]
What I found so interesting is that it worked completely fine with the try.nextcloud account (I even tested with a shared board). Can I help you with any information on my server side? Or is there anything that could be "tuned" e.g in the nextcloud deck app?

**Comment by desperateCoder** [comment]
In fact, I had the same issue a few days ago, I think my instance could reproduce this. We'll analyze this, for now, unfortunately I only can guess what's happening... But thanks for your cooperation so far! We might return to the offer!

**Comment by stefan-niedermann** [comment]
I have also been able to reproduce the issue (with a build which already contains queued sync).

I also checked that we only [`observeOnce`](https://github.com/stefan-niedermann/nextcloud-deck/blob/master/app/src/main/java/it/niedermann/nextcloud/deck/ui/card/EditActivity.java#L198). Currently i suspect the `EditActivity` to not finish proper after creating the first card, but if this was the case, it should create linear `n` new cards with each new card (which is not the case).

Hmm 🤔  guess we need some debugging here. I really wished we already had Android Studio 4.1 with the new Database panel.... 😞

**Comment by stefan-niedermann** [comment]
I have also been able to reproduce the issue (with a build which already contains queued sync).

I also checked that we only [`observeOnce`](https://github.com/stefan-niedermann/nextcloud-deck/blob/master/app/src/main/java/it/niedermann/nextcloud/deck/ui/card/EditActivity.java#L198). Currently i suspect the `EditActivity` to not finish proper after creating the first card, but if this was the case, it should create linear `n` new cards with each new card (which is not the case).

Hmm 🤔  guess we need some debugging here. I really wished we already had Android Studio 4.1 with the new Database panel.... 😞

**Comment by stefan-niedermann** [comment]
It seem to work properly if i go offline and create many cards. Going online again syncs them up as expected. @tuximail can you confirm this behavior?

So maybe `createFullCard` triggers a network request (if online) and in parallel creates the card locally or so? @desperateCoder

**Comment by tuximail** [comment]
I can confirm the behavior in airplane/offline mode. After coming back online the newly created cards were synced properly...

**Comment by stefan-niedermann** [comment]
@tuximail can you please test this again as soon as `1.10.0` has been released? This should be fixed starting with this version.

If it still occurs with `1.10.0` or higher, just reopen this issue please :)

Thank you for reporting the issue!
